crates/oxc_minifier/src/ast_passes/peephole_fold_constants.rs

```diff
@@ -157,8 +157,19 @@ impl<'a> PeepholeFoldConstants {
                 Some(ctx.ast.move_expression(&mut expr.argument))
             }
             // `+1` -> `1`
-            UnaryOperator::UnaryPlus if expr.argument.is_number() => {
-                Some(ctx.ast.move_expression(&mut expr.argument))
+            UnaryOperator::UnaryPlus => {
+                if expr.argument.is_number() {
+                    Some(ctx.ast.move_expression(&mut expr.argument))
+                } else if let Expression::UnaryExpression(un_expr) = &mut expr.argument {
+                    // handle +-
+                    if un_expr.operator == UnaryOperator::UnaryNegation {
```

Comment:
You can try the https://doc.rust-lang.org/std/primitive.bool.html#method.then method

Reply:
Refactored.